### PR TITLE
More recipes for BERT 3-, 6-layer models

### PR DESCRIPTION
**Pull request by natuan**
Recipes for BERT models with 3 layers (70%, 80%, 90%, 95% sparsity) and 6 layers (70% sparsity).

### CODE REVIEW

**Review by bfineran — DISMISSED** [review]

File: integrations/huggingface-transformers/recipes/bert-base-3layers_prune70.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    inter_func: cubic
+    update_frequency: *pruning_update_frequency
+    leave_enabled: True
+    mask_type: unstructured
+    log_types: __ALL__
+---
+
+# BERT Model with Dropped and Pruned Encoder Layers
+
+This recipe defines a dropping and pruning strategy to sparsify three encoder layers of a BERT model at 70% sparsity. It was used together with knowledge distillation to create sparse model that achives 90.5% recovery from the F1 metric of the baseline model on the SQuAD dataset. (We use the teacher model fine tuned for 2 epochs as the baseline for comparison.)
```
Comment (jeanniefinks) [inline integrations/huggingface-transformers/recipes/bert-base-3layers_prune70.md:63]:
```suggestion
This recipe defines a dropping and pruning strategy to sparsify three encoder layers of a BERT model at 70% sparsity. It was used together with knowledge distillation to create a sparse model that achieves 90.5% recovery from the F1 metric of the baseline model on the SQuAD dataset. (We use the teacher model fine-tuned for 2 epochs as the baseline for comparison.)
```

File: integrations/huggingface-transformers/recipes/bert-base-3layers_prune80.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    inter_func: cubic
+    update_frequency: *pruning_update_frequency
+    leave_enabled: True
+    mask_type: unstructured
+    log_types: __ALL__
+---
+
+# BERT Model with Dropped and Pruned Encoder Layers
+
+This recipe defines a dropping and pruning strategy to sparsify three encoder layers of a BERT model at 80% sparsity. It was used together with knowledge distillation to create sparse model that achives 89% recovery from the F1 metric of the baseline model on the SQuAD dataset. (We use the teacher model fine tuned for 2 epochs as the baseline for comparison.)
```
Comment (jeanniefinks) [inline integrations/huggingface-transformers/recipes/bert-base-3layers_prune80.md:63]:
```suggestion
This recipe defines a dropping and pruning strategy to sparsify three encoder layers of a BERT model at 80% sparsity. It was used together with knowledge distillation to create a sparse model that achieves 89% recovery from the F1 metric of the baseline model on the SQuAD dataset. (We use the teacher model fine-tuned for 2 epochs as the baseline for comparison.)
```

File: integrations/huggingface-transformers/recipes/bert-base-3layers_prune90.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    inter_func: cubic
+    update_frequency: *pruning_update_frequency
+    leave_enabled: True
+    mask_type: unstructured
+    log_types: __ALL__
+---
+
+# BERT Model with Dropped and Pruned Encoder Layers
+
+This recipe defines a dropping and pruning strategy to sparsify three encoder layers of a BERT model at 90% sparsity. It was used together with knowledge distillation to create sparse model that achives 86% recovery from the F1 metric of the baseline model on the SQuAD dataset. (We use the teacher model fine tuned for 2 epochs as the baseline for comparison.)
```
Comment (jeanniefinks) [inline integrations/huggingface-transformers/recipes/bert-base-3layers_prune90.md:63]:
```suggestion
This recipe defines a dropping and pruning strategy to sparsify three encoder layers of a BERT model at 90% sparsity. It was used together with knowledge distillation to create a sparse model that achieves 86% recovery from the F1 metric of the baseline model on the SQuAD dataset. (We use the teacher model fine-tuned for 2 epochs as the baseline for comparison.)
```

File: integrations/huggingface-transformers/recipes/bert-base-3layers_prune95.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    inter_func: cubic
+    update_frequency: *pruning_update_frequency
+    leave_enabled: True
+    mask_type: unstructured
+    log_types: __ALL__
+---
+
+# BERT Model with Dropped and Pruned Encoder Layers
+
+This recipe defines a dropping and pruning strategy to sparsify three encoder layers of a BERT model at 95% sparsity. It was used together with knowledge distillation to create sparse model that achives 83% recovery from the F1 metric of the baseline model on the SQuAD dataset. (We use the teacher model fine tuned for 2 epochs as the baseline for comparison.)
```
Comment (jeanniefinks) [inline integrations/huggingface-transformers/recipes/bert-base-3layers_prune95.md:63]:
```suggestion
This recipe defines a dropping and pruning strategy to sparsify three encoder layers of a BERT model at 95% sparsity. It was used together with knowledge distillation to create a sparse model that achieves 83% recovery from the F1 metric of the baseline model on the SQuAD dataset. (We use the teacher model fine-tuned for 2 epochs as the baseline for comparison.)
```

File: integrations/huggingface-transformers/recipes/bert-base-6layers_prune70.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+    inter_func: cubic
+    update_frequency: *pruning_update_frequency
+    leave_enabled: True
+    mask_type: unstructured
+    log_types: __ALL__
+---
+
+# BERT Model with Dropped and Pruned Encoder Layers
+
+This recipe defines a dropping and pruning strategy to sparsify six encoder layers of a BERT model at 70% sparsity. It was used together with knowledge distillation to create sparse model that achives 83% recovery from the F1 metric of the baseline model on the SQuAD dataset. (We use the teacher model fine tuned for 2 epochs as the baseline for comparison.)
```
Comment (jeanniefinks) [inline integrations/huggingface-transformers/recipes/bert-base-6layers_prune70.md:63]:
```suggestion
This recipe defines a dropping and pruning strategy to sparsify six encoder layers of a BERT model at 70% sparsity. It was used together with knowledge distillation to create a sparse model that achieves 83% recovery from the F1 metric of the baseline model on the SQuAD dataset. (We use the teacher model fine-tuned for 2 epochs as the baseline for comparison.)
```

File: integrations/huggingface-transformers/recipes/bert-base-3layers_prune70.md (hunk 1 above)
Comment (natuan) [reply integrations/huggingface-transformers/recipes/bert-base-3layers_prune70.md]: Fix

File: integrations/huggingface-transformers/recipes/bert-base-3layers_prune80.md (hunk 2 above)
Comment (natuan) [reply integrations/huggingface-transformers/recipes/bert-base-3layers_prune80.md]: Fix

**Review by bfineran — APPROVED** [review]

**Review by jeanniefinks — APPROVED** [review]

**Review by markurtz — APPROVED** [review]

**Pull request merged**